### PR TITLE
Add All Tracks view and Download All Tracks functionality

### PR DESCRIPTION
**Pull request by Cj1m**
Add All Tracks view to allow users to see all their tracks from singles and albums
Add Download All functionality
Add album art to TrackPopupMenu

### CODE REVIEW

File: src/screens/Music/stacks/Singles.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { useCallback, useEffect } from 'react';
+import { useRoute, RouteProp } from '@react-navigation/native';
```
Comment (leinelissen) [inline src/screens/Music/stacks/Singles.tsx:2]: This import (and a bunch of others in this file) is unused. Please remove any unused imports. Please check the output of `npm run lint` as well.

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    border-bottom-width: 1px;
+    flex-direction: row;
+
+    ${props => props.isPlaying && css`
+        margin: 0 -20px;
+        padding: 15px 24px;
+    `}
+`;
+
+const SingleInformationContainer = styled.View`
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:57]: This variable is unused.

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    }, [playTracks, trackIds]);
+    const longPressTrack = useCallback((index: number) => { 
+        navigation.navigate('TrackPopupMenu', { trackId: trackIds[index] }); 
+    }, [navigation, trackIds]);
+    const downloadAllTracks = useCallback(() => {
+        trackIds.forEach((trackId) => dispatch(queueTrackForDownload(trackId)));
+    }, [dispatch, trackIds]);
+
+    return (
+        <ScrollView
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:107]: Please use a FlatList, as huge music libraries have a ton of trouble with rendering large libraries.

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                        onPress={selectTrack}
+                        onLongPress={longPressTrack}
+                    >
+                        <TrackContainer
+                            isPlaying={currentTrack?.backendId === trackId || false}
+                            style={[defaultStyles.border, currentTrack?.backendId === trackId || false ? defaultStyles.activeBackground : null ]}
+                        >
+                            {
+                                tracks[trackId]?.AlbumId ?
+                                    <SingleImage source={{ uri: getImage(String(tracks[trackId]?.AlbumId)) }} style={defaultStyles.imageBackground} /> :
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:131]: I think it works better for the design if you make the image a bit smaller (look at the Search screen for an example). This leaves a bit more tracks in view, which makes it easier to scroll through a large library.

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                            }
+                            
+                            <View style={[{flexDirection:'column', marginLeft: 20, marginRight:10, flexShrink: 1 }]}>
+                                <Text
+                                    style={[currentTrack?.backendId === trackId
+                                        ? { color: THEME_COLOR, fontWeight: '700' }
+                                        : defaultStyles.text,
+                                        {flexWrap: 'wrap'},
+                                    ]}
+                                    numberOfLines={2}
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:142]: It's probably best for performance to keep this to a single line

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    artistTrackText:{
+        opacity: 0.5,
+        display: 'flex'
+    }
+});
+
+const TrackContainer = styled.View<{isPlaying: boolean}>`
+    padding: 5px 4px;
+    border-bottom-width: 1px;
+    flex-direction: row;
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:49]: If you add `align-items: center;` here, all the individual parts will be nicely aligned.

File: src/screens/Music/stacks/components/SinglesListView.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                                </Text>
+                            </View>
+                            <View style={{ marginLeft: 'auto'}}>
+                                <DownloadIcon trackId={trackId} />
+                            </View>
+                        </TrackContainer>
+                    </TouchableHandler>
+                )}
+                <WrappableButtonRow style={{ marginTop: 24 }}>
+                    <WrappableButton
```
Comment (leinelissen) [inline src/screens/Music/stacks/components/SinglesListView.tsx:164]: It's probably best to put this up top, as it takes a really long time to scroll down (and find it) if your library is huge.

File: src/store/music/index.ts [hunk 8]
```diff
@@ -44,7 +45,7 @@ export const initialState: State = {
         ...trackAdapter.getInitialState(),
         isLoading: false,
         byAlbum: {},
-        byPlaylist: {},
+        byPlaylist: {}
```
Comment (leinelissen) [inline src/store/music/index.ts:48]: The trailing comma is supposed to be here 😅

**Review by leinelissen — CHANGES_REQUESTED** [review]
Hey @Cj1m, thanks for your amazing contribution! I see you were able to figure out most of the codebase by yourself. I've put together some suggestions that should make your contribution perform well and be more consistent with the rest of the UI. If you could take a look at integrating the suggested changes, I'm happy to merge and release it.

**Comment by nbabanov** [comment]
@leinelissen  I can fork @Cj1m's repo and fix the mentioned issues. Would then accept the new MR(PR)?

**Comment by leinelissen** [comment]
Yes, that would be greatly appreciated!

**Pull request closed without merging**